src/object/obj_tx.c

```diff
@@ -1917,14 +1960,45 @@ dc_tx_restart(tse_task_t *task)
 			tx->tx_status);
 		rc = -DER_NO_PERM;
 	} else {
+		uint32_t backoff;
+
 		dc_tx_cleanup(tx);
 
-		tx->tx_status = TX_OPEN;
-		tx->tx_epoch.oe_value = 0;
 		if (tx->tx_epoch_task != NULL) {
 			tse_task_decref(tx->tx_epoch_task);
 			tx->tx_epoch_task = NULL;
 		}
+
+		backoff = d_backoff_seq_next(&tx->tx_backoff_seq);
+		if (backoff == 0) {
+			/* No child task is required; reopen the TX now. */
+			dc_tx_restart_reopen(tx);
+		} else {
+			tse_task_t *child;
+
+			/*
+			 * Schedule a delayed child task to implement the
+			 * backoff.
+			 */
+			rc = tse_task_create(dc_tx_restart_reopen_task,
```

Comment:
I’m wondering if we need a child task here? Can we reinit the original task with delay?

Reply:
@mchaarawi, I thought you and Di would prefer a child task. Changed to reinit. (In retrospect, the child task version looks a bit easier to understand.)